osu.Game/Input/Bindings/DatabasedKeyBindingContainer.cs

```diff
@@ -42,24 +48,34 @@ public DatabasedKeyBindingContainer(RulesetInfo ruleset = null, int? variant = n
                 throw new InvalidOperationException($"{nameof(variant)} can not be null when a non-null {nameof(ruleset)} is provided.");
         }
 
-        [BackgroundDependencyLoader]
-        private void load(KeyBindingStore keyBindings)
-        {
-            store = keyBindings;
-        }
-
         protected override void LoadComplete()
         {
+            if (ruleset == null || ruleset.ID.HasValue)
```

Comment:
This condition's weird. What's it actually doing? It seems like the only case which isn't handled is `ruleset.ID == null`, but
a) It shouldn't cause any problems as far as I can see (see `ruleset?.Id` below).
b) If this is truly an unsupported case, shouldn't it throw in the ctor?

Reply:
This comes from the original logic which is still present on `master`. I'm willing to take a look and figure out why it is written this way in this PR if you'd prefer. Do you see this as a blocker?

Reply:
Ah, this is the check inside `ReloadMappings`? I see. Looks like it was done that way intentionally so it's fine to leave it.

Reply:
I'll look into this at some point, and potentially push a change to this PR if it's not merged by the time I do.